src/python/grpcio_observability/grpc_observability/_open_telemetry_observability.py

```diff
@@ -80,26 +112,39 @@ def __init__(
 
         self.exporter = _OpenTelemetryExporterDelegator(_plugins)
 
+    def __enter__(self):
```

Comment:
It looks like this has a lot in common with the implementation of `start_open_telemetry_observability`. The only real difference seems to be that here, we _know_ which instance of `OpenTelemetryObservability` we want to set `_OPEN_TELEMETRY_OBSERVABILITY` to and in the case of `start_open_telemetry_observability`, we need to create a new one. How about we have a `def _start_open_telemetry_observability(o: OpenTelemetryObservability, plugins: Optional[Iterable[OpenTelemetryPlugin]])` function that both share?